internal/pkg/cli/job_logs.go

```diff
@@ -151,38 +157,43 @@ func (o *jobLogsOpts) Ask() error {
 	return o.validateAndAskJobEnvName()
 }
 
-func (o *jobLogsOpts) validateOrAskApp() error {
-	if o.appName != "" {
-		_, err := o.configStore.GetApplication(o.appName)
+// Execute outputs logs of the job.
+func (o *jobLogsOpts) Execute() error {
+	if err := o.initLogsSvc(); err != nil {
 		return err
 	}
-	app, err := o.sel.Application(jobAppNamePrompt, svcAppNameHelpPrompt)
-	if err != nil {
-		return fmt.Errorf("select application: %w", err)
+	eventsWriter := logging.WriteHumanLogs
+	if o.shouldOutputJSON {
+		eventsWriter = logging.WriteJSONLogs
+	}
+	var limit *int64
+	if o.limit != 0 {
+		limit = aws.Int64(int64(o.limit))
 	}
-	o.appName = app
-	return nil
-}
 
-func (o *jobLogsOpts) validateAndAskJobEnvName() error {
-	if o.envName != "" {
-		if _, err := o.getTargetEnv(); err != nil {
-			return err
-		}
+	// By default, only display the logs of the last execution of the job.
+	logStreamLimit := defaultJobLogExecutionLimit
+	if o.last != 0 {
+		logStreamLimit = o.last
 	}
 
-	if o.name != "" {
-		if _, err := o.configStore.GetJob(o.appName, o.name); err != nil {
-			return err
-		}
+	if o.includeStateMachineLogs {
+		logStreamLimit *= 2
 	}
 
-	deployedJob, err := o.sel.DeployedJob(jobLogNamePrompt, jobLogNameHelpPrompt, o.appName, selector.WithEnv(o.envName), selector.WithName(o.name))
+	err := o.logsSvc.WriteLogEvents(logging.WriteLogEventsOpts{
+		Follow:                  o.follow,
+		Limit:                   limit,
+		EndTime:                 o.endTime,
+		StartTime:               o.startTime,
+		TaskIDs:                 o.taskIDs,
+		OnEvents:                eventsWriter,
+		LogStreamLimit:          logStreamLimit,
```

Comment:
I realized I've never noticed this parameter existed before :homer-disappearing !  If `logStreamLimit`, is it guaranteed that the one and only one log stream that gets fetched is the log stream that's produced by the latest call? Could it be the log stream that's produced by the second lastest call? (sorry for the repetition of words 😂 )

Reply:
It's the most recent that Cloudwatch has to offer; I don't know that we can control the delay between a container coming up and the log stream for that container appearing. But I assume that there will be few times when someone needs to immediately view the logs from the last execution less than 10 seconds after the job is triggered, and this case can be covered by
```sh
copilot job run && copilot job logs --follow
``` 
or eventually
```sh
copilot job run --follow
```